numpy/lib/tests/test_function_base.py

```diff
@@ -3104,6 +3108,58 @@ def test_no_p_overwrite(self):
         np.quantile(np.arange(100.), p, interpolation="midpoint")
         assert_array_equal(p, p0)
 
+    def test_quantile_monotonic(self):
+        # GH 14685
+        # test that the return value of quantile is monotonic if p0 is ordered
+        p0 = np.arange(0, 1, 0.01)
+        quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9, 8, 8, 7]) * 0.1, p0)
+        equals_sorted = np.sort(quantile) == quantile
+        assert equals_sorted.all()
+
+    @hypothesis.given(arr=arrays(dtype=np.float, shape=st.integers(min_value=3, max_value=1000),
+                                 elements=st.floats(allow_infinity=False, allow_nan=False)))
+    def test_quantile_monotonic_hypo(self, arr):
+        p0 = np.arange(0, 1, 0.01)
+        quantile = np.quantile(arr, p0)
+        equals_sorted = np.sort(quantile) == quantile
+        assert equals_sorted.all()
+
+
+class TestLerp:
+    @hypothesis.given(t0=st.floats(allow_nan=False, allow_infinity=False,
+                                   min_value=0, max_value=1),
+                      t1=st.floats(allow_nan=False, allow_infinity=False,
+                                   min_value=0, max_value=1),
+                      a=st.floats(allow_nan=False, allow_infinity=False,
+                                  width=32),
+                      b= st.floats(allow_nan=False, allow_infinity=False,
+                                   width=32))
+    def test_lerp_monotonic(self, t0, t1, a, b):
+        assert (np.lib.function_base._lerp(a, b, t1) -
+                np.lib.function_base._lerp(a, b, t0)) * (b - a) * (t1 - t0) >= 0
```

Comment:
I'm a bit worried this test is unsafe because these multiplications could cancel to 0 for the extreme cases that we're worried about. Can you write it as the safer:
```suggestion
        l0 = np.lib.function_base._lerp(a, b, t0)
        l1 = np.lib.function_base._lerp(a, b, t1)
        if t0 == t1 or a == b:
            assert l0 == l1  # uninteresting
        elif (t0 < t1) == (a < b):
            assert l0 < l1
        else:
            assert l0 > l1
```